runtime/termscounter_test.go

```diff
@@ -0,0 +1,147 @@
+package runtime_test
+
+import (
+	"go.etcd.io/gofail/runtime"
+	"testing"
+)
+
+var __fp_ExampleString *runtime.Failpoint
+
+// check if failpoint is initialized as gofail
+// tests can clear global variables of runtime packages
+func initFP() {
+	if __fp_ExampleString == nil {
+		__fp_ExampleString = runtime.NewFailpoint("ExampleString") //nolint:stylecheck
+	}
+}
+
+func TestTermsCounter(t *testing.T) {
+	testcases := []struct {
+		name              string
+		fp                string
+		failpointTerm     string
+		runBeforeEnabling int
+		runAfterEnabling  int
+		wantCount         int
+	}{
+		{
+			name:             "Terms limit Failpoint",
+			fp:               "ExampleString",
+			failpointTerm:    `10*sleep(10)->1*return("abc")`,
+			runAfterEnabling: 12,
+			// This refers to the number of term executions and is
+			// independent of caller execution frequency. E.g., Run
+			// failpoint actions only 11 times even if we have a greater
+			// number of callsite executions (12 in this case)
```

Comment:
I don't think this explains why `run failpoint actions` != `callsite executions`

Reply:
From my understanding, the reason is to qualify the number of times the failpoint actions happen. So if the callsite is hit X number of times but you want to restrict term executions to Y (typically Y < X), then you either use a hard count or a probability to restrict. Since the original author is not active I am not sure if this is indeed the reason. Maybe @ahrtr can comment.

Reply:
> the reason is to qualify the number of times the failpoint actions happen

Yes, we want to count number of times failpoint i executed.

> So if the callsite is hit X number of times but you want to restrict term executions to Y (typically Y < X), then you either use a hard count or a probability to restrict

Don't understand

Please double check the reason and document it as a comment. We shouldn't merge code that we don't understand.

Reply:
> So if the callsite is hit X number of times but you want to restrict term executions to Y (typically Y < X), then you either use a hard count or a probability to restrict

> Don't understand

This is implemented by mods([code](https://github.com/etcd-io/gofail/blob/898fcaf00ea0c5b6a27f879ff1952f6fb391d93a/runtime/terms.go#LL104C3-L104C25)). The terms used in the example (`10*sleep(10)->1*return("abc")`) have mods in them. You can specify a floating point probabilityand the term actions will be executed only % of callsite executions. If you remove mods i.e., not have int or float in the terms, then the number of term executions will equal callsite executions. This example tests mods and so the number of callsite executions != failpoint actions.

Reply:
Note: Don't worry about explaining this to me. Just improve the comment to explain the test case as accurate as possible.

Reply:
I have updated the comment. Please check.